FantasmoSDK/app/src/main/java/com/fantasmo/sdk/config/RemoteConfig.kt

```diff
@@ -35,19 +37,50 @@ class RemoteConfig {
         lateinit var remoteConfig: Config
 
         /**
-         * Overrides the default config with a remote Config
+         * Updates current `Config`
+         * @param context Application context
+         * @param jsonString `Json` read from the request response
          */
-        fun updateConfig(jsonString: String) {
-            Log.i(TAG, "Received Remote Config.")
-            remoteConfig = getConfigFromJSON(jsonString)
+        fun updateConfig(context: Context, jsonString: String) {
+            val config = getConfigFromJSON(jsonString)
+            remoteConfig = if (config == null) {
+                getConfig(context)!!
+            } else {
+                Log.i(TAG, "Received Valid Remote Config.")
+                try {
+                    val file = File(context.filesDir, fileName)
+                    val fileWriter = FileWriter(file)
+                    val bufferedWriter = BufferedWriter(fileWriter)
+                    bufferedWriter.write(jsonString)
+                    bufferedWriter.close()
+                    Log.i(TAG, "Successfully saved new remote config.")
+                } catch (e: IOException) {
+                    Log.e(TAG, "Remote Config File write failed: $e.")
+                }
+                config
```

Comment:
We are writing the remote config to file. If the received remote config is valid (!=null) then we stored in a json file. If that writing fails we still use it because it's already in memory. Perhaps I should clear better this.